lib/create-promise.js

```diff
@@ -0,0 +1,29 @@
+'use strict'
+
+/**
+ * @typedef PromiseObject
+ * @property {Promise} promise
+ * @property {PromiseConstructor["resolve"]} resolve
+ * @property {PromiseConstructor["reject"]} reject
+ */
+
+/**
+ * @returns {PromiseObject}
+ */
+function createPromise () {
+  /**
+     * @type {PromiseObject}
+     */
+  const obj = {}
```

Comment:
```suggestion
  const obj = {
    resolve: null,
    reject: null,
    promise: null
  }
```

This is slightly better.